statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,254 @@
+            aic : Akaike
+            fpe : Final prediction error
+            hqic : Hannan-Quinn
+            bic : Bayesian a.k.a. Schwarz
+        deterministic terms, str {"", "c", "lt", "s"}
+            "" - no deterministic terms
+            "c" - constant
+            "lt" - linear trend
+            "s" - seasonal terms
+            Combinations of these are possible (e.g. "clt" for linear trend 
+            with intercept)
+
+        Notes
+        -----
+        Lutkepohl pp. 146-153
+
+        Returns
+        -------
+        est : VARResults
+        """#TODO: docstring + implementation
+        
+        #TODO: trend
+        
+        # select number of lags (=p-1)
+        if ic is None:
+            if max_diff_lags is None:
+                diff_lags = 1
+            else:
+                diff_lags = max_diff_lags
+        else:
+            selections = self.select_order(max_diff_lags=max_diff_lags, verbose=verbose)
+            if ic not in selections:
+                raise Exception("%s not recognized, must be among %s"
+                                % (ic, sorted(selections)))
+            diff_lags = selections[ic]
+            if verbose:
+                print('Using %d based on %s criterion' %  (diff_lags, ic))
+        
+        # estimate parameters
+        if method=='ls':
+            return self._estimate_vecm_ls(diff_lags, deterministic_terms)
+        elif method == 'egls':
+            if coint_rank is None:
+                coint_rank = 1
+            return self._estimate_vecm_egls(diff_lags, deterministic_terms, coint_rank)
+        elif method == 'ml':
+            if coint_rank is None:
+                coint_rank = 1
+            return self._estimate_vecm_ml(diff_lags, deterministic_terms, coint_rank)
+        else:
+            raise Exception("%s not recognized, must be among %s"
+                            % (method, ("ls", "egls", "ml")))
+        
+
+    def estimation_matrices(self, diff_lags, deterministic_terms):
+        p = diff_lags+1
+        y_all = self.endog.T#######################superclass turning DataFrame into ndarray?
+        K = y_all.shape[0]
+        
+        y_1_T = y_all[:, p:]
+        T = y_1_T.shape[1]
+        
+        Delta_y_all = np.diff(y_all)
+        Delta_y_1_T = Delta_y_all[:, p-1:]
+        
+        y_minus1 = y_all[:, p-1:-1]
+        if 'lt' in deterministic_terms:
+            y_minus1 = np.vstack((y_minus1, np.arange(T)))
+        
+        Delta_x = np.zeros((diff_lags*K, T))
+        for j in range(Delta_x.shape[1]):
+            Delta_x[:, j] = Delta_y_all[:, j+p-2:None if j-1<0 else j-1:-1].T.reshape(K*(p-1))
+        if 'c' in deterministic_terms:
+            Delta_x = np.vstack((Delta_x, np.ones(T)))
+        if 's' in deterministic_terms: # TODO: How many seasons??
+            seasons = np.zeros((3, Delta_x.shape[1])) # 3 = 4-1 (4=number_of_seasons)
+            for i in range(seasons.shape[0]):
+                seasons[i, i::4] = i+1
```

Comment:
AFAICS, this should be assigning 1, i.e.
`seasons[i, i::4] = 1`

This should fix the test failure for non-seasonal parameters, even though it might not be the same coding that JMulti is using, and it still leaves periodicity hardcoded at 4.